lib/ts/recipe/oauth2/types.ts

```diff
@@ -0,0 +1,298 @@
+    };
+};
+
+export type TypeNormalisedInput = {
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+        apis: (originalImplementation: APIInterface, builder?: OverrideableBuilder<APIInterface>) => APIInterface;
+    };
+};
+
+export type APIOptions = {
+    recipeImplementation: RecipeInterface;
+    config: TypeNormalisedInput;
+    recipeId: string;
+    isInServerlessEnv: boolean;
+    req: BaseRequest;
+    res: BaseResponse;
+};
+
+export type APIInterface = {
+    tokenPOST(input: {
+        clientId: string;
+        grantType: GrantType;
+        clientSecret?: string;
+        code?: string;
+        codeVerifier?: string;
+        refreshToken?: string;
+        redirectUri?: string;
+        scope: string[];
+
+        options: APIOptions;
+        userContext: any;
+    }): Promise<
+        | {
+              token_type: "Bearer";
+              access_token: string;
+              refresh_token?: string;
+              id_token?: string;
+              expires_in: number;
+          }
+        | GeneralErrorResponse
+    >;
+    // The OpenId spec states that we have to support both POST and GET
+    authorizeGET(input: {
+        clientId: string;
+        responseType: ResponseType[];
+        redirectUri: string;
+        scope: string[];
+        state?: string;
+
+        // PKCE
+        codeChallenge?: string;
+        codeChallengeMethod?: CodeChallengeMethod;
+
+        // OpenIdConnect
+        prompt?: OAuthPromptType;
+        nonce?: string;
+
+        session?: SessionContainer;
+
+        options: APIOptions;
+        userContext: any;
+    }): Promise<
+        | {
+              redirectUrl: string; // This will end up as a 302, encoding both error and success
+          }
+        | GeneralErrorResponse
+    >;
+    authorizePOST(input: {
+        clientId: string;
+        responseType: ResponseType[]; // add enum
+        redirectUri: string;
+        scope: string[];
+        state?: string;
+
+        // PKCE
+        codeChallenge?: string;
+        codeChallengeMethod?: CodeChallengeMethod;
+
+        // OpenIdConnect
+        prompt?: OAuthPromptType;
+        nonce?: string;
+
+        session?: SessionContainer;
+
+        options: APIOptions;
+        userContext: any;
+    }): Promise<
+        | {
+              redirectUrl: string; // This will end up as a 302, encoding both error and success
+          }
+        | GeneralErrorResponse
+    >;
+
+    userInfoGET(input: {
+        oauth2AccessToken: string;
+        options: APIOptions;
+        userContext: any;
+    }): Promise<JSONObject | { status: "UNAUTHORISED"; message: string }>;
+};
+
+export type RecipeInterface = {
+    getUserInfo(input: {
+        oauth2AccessToken: string;
+
+        userContext: any;
+    }): Promise<JSONObject>;
```

Comment:
is there a core endpoint that's required for this? Or else how will this get the info?

Reply:
No, this is done in the BE SDK. Basically it uses the session handle (and the user id). The result of this basically built the same way as the id token.